lib/diplomat/value.ex

```diff
@@ -14,85 +19,102 @@ defmodule Diplomat.Value do
 
   def new(val), do: %__MODULE__{value: val}
 
-  def from_proto(%PbVal{boolean_value: val}) when is_boolean(val),
-    do: new(val)
-  def from_proto(%PbVal{integer_value: val}) when is_integer(val),
-    do: new(val)
-  def from_proto(%PbVal{double_value: val}) when is_float(val),
-    do: new(val)
-  def from_proto(%PbVal{string_value: val}) when is_binary(val),
-    do: new(val)
-  def from_proto(%PbVal{blob_value: val}) when is_bitstring(val),
-    do: new(val)
-  def from_proto(%PbVal{timestamp_microseconds_value: val}) when is_integer(val),
-    do: new(val) # need to convert this to a timestamp of some sort
-  def from_proto(%PbVal{key_value: val}) when not is_nil(val) do
+  def from_proto(%PbVal{value_type: {:boolean_value, val}}) when is_boolean(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:integer_value, val}}) when is_integer(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:double_value, val}}) when is_float(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:string_value, val}}) when is_binary(val) do
+    new(val)
+  end
+  def from_proto(%PbVal{value_type: {:blob_value, val}}) when is_bitstring(val) do
+    new(val)
+  end
```

Comment:
I change `from_proto/1` function as you suggested

For `proto/1` function, I think we should keep the format

```elixir
def proto(val) do
end
```

instead of 

```elixir
def proto(val),
   do: 
```

because there is some case 

```
  def proto(val) when is_binary(val) do
    case String.valid?(val) do
      true -> PbVal.new(value_type: {:string_value, val})
      false-> PbVal.new(value_type: {:blob_value, val})
    end
  end
```

and I think we should keep all these functions should have the same format

I would love to hear your opinion about this thing.